homeassistant/components/music_assistant/media_player.py

```diff
@@ -0,0 +1,729 @@
+        album: str | None = None,
+        media_type: str | None = None,
+    ) -> MediaItemType | None:
+        """Try to find a media item (such as a playlist) by name."""
+        # pylint: disable=too-many-nested-blocks
+        searchname = name.lower()
+
+        funcs: Sequence[Callable] = (
+            self.mass.music.get_library_playlists,
+            self.mass.music.get_library_radios,
+            self.mass.music.get_library_tracks,
+            self.mass.music.get_library_albums,
+            self.mass.music.get_library_artists,
+        )
+
+        library_functions = [
+            x for x in funcs if not media_type or media_type.lower() in x.__name__
+        ]
+        # prefer (exact) lookup in the library by name
+        for func in library_functions:
+            result = await func(search=searchname)
+            for item in result:
+                # handle optional artist filter
+                if (
+                    artist
+                    and (artists := getattr(item, "artists", None))
+                    and not any(x for x in artists if x.name.lower() == artist.lower())
+                ):
+                    continue
+                # handle optional album filter
+                if (
+                    album
+                    and (item_album := getattr(item, "album", None))
+                    and item_album.name.lower() != album.lower()
+                ):
+                    continue
+                if searchname == item.name.lower():
+                    return item  # type: ignore[no-any-return]
+        # nothing found in the library, fallback to global search
+        search_name = name
+        if album and artist:
+            search_name = f"{artist} - {album} - {name}"
+        elif album:
+            search_name = f"{album} - {name}"
+        elif artist:
+            search_name = f"{artist} - {name}"
+        result = await self.mass.music.search(
+            search_query=search_name,
+            media_types=[MediaType(media_type)] if media_type else MediaType.ALL,
+            limit=5,
+        )
+        for results in (
+            result.tracks,
+            result.albums,
+            result.playlists,
+            result.artists,
+            result.radio,
+        ):
+            for item in results:
+                # simply return the first item because search is already sorted by best match
+                return item  # type: ignore[no-any-return]
+        return None
+
+    def _update_media_image_url(
+        self, player: Player, queue: PlayerQueue | None
+    ) -> None:
+        """Update image URL for the active queue item."""
+        if queue is None or queue.current_item is None:
+            self._attr_media_image_url = None
+            return
+        if image_url := self.mass.get_media_item_image_url(queue.current_item):
+            self._attr_media_image_remotely_accessible = (
+                self.mass.server_url not in image_url
+            )
```

Comment:
So we mean to say, if the image comes from an external source, we set it to True?